index.js

```diff
@@ -39,355 +32,291 @@ module.exports = Store;
+    this.set(key, unique(flatten(...arr, ...rest)));
+    return this;
   }
-  return val;
-};
 
-/**
- * Persist the store to disk.
- *
- * ```js
- * store.save();
- * ```
- * @param {String} `dest` Optionally define an alternate destination file path.
- * @api public
- */
+  /**
+   * Get the stored `value` of `key`, or return the entire store
+   * if no `key` is defined.
+   *
+   * ```js
+   * store.set('a', {b: 'c'});
+   * store.get('a');
+   * //=> {b: 'c'}
+   *
+   * store.get();
+   * //=> {b: 'c'}
+   * ```
+   *
+   * @name .get
+   * @param  {String} `key`
+   * @return {any} The value to store for `key`.
+   * @api public
+   */
+
+  get(key) {
+    return key ? get(this.data, key) : this.data;
+  }
 
-Store.prototype.save = function(dest) {
-  this.data = this.cache;
-  writeJson(dest || this.path, this.cache, this.options.indent);
-  return this;
-};
+  /**
+   * Returns `true` if the specified `key` has a value.
+   *
+   * ```js
+   * store.set('a', 'b');
+   * store.set('c', null);
+   * store.has('a'); //=> true
+   * store.has('c'); //=> false
+   * store.has('d'); //=> false
+   * ```
+   * @name .has
+   * @param  {String} `key`
+   * @return {Boolean} Returns true if `key` has
+   * @api public
+   */
+
+  has(key) {
+    assert.equal(typeof key, 'string', 'expected key to be a string');
+    return typeof get(this.data, key) !== 'undefined';
+  }
 
-/**
- * Clear in-memory cache.
- *
- * ```js
- * store.clear();
- * ```
- * @api public
- */
+  /**
+   * Returns `true` if the specified `key` exists.
+   *
+   * ```js
+   * store.set('a', 'b');
+   * store.set('b', false);
+   * store.set('c', null);
+   * store.set('d', true);
+   *
+   * store.hasOwn('a'); //=> true
+   * store.hasOwn('b'); //=> true
+   * store.hasOwn('c'); //=> true
+   * store.hasOwn('d'); //=> true
+   * store.hasOwn('foo'); //=> false
+   * ```
+   *
+   * @param  {String} `key`
+   * @return {Boolean} Returns true if `key` exists
+   * @api public
+   */
+
+  hasOwn(key) {
+    assert.equal(typeof key, 'string', 'expected key to be a string');
+    return hasOwn(this.data, key);
+  }
 
-Store.prototype.clear = function() {
-  this.cache = {};
-  this.data = {};
-  return this;
-};
+  /**
+   * Delete `keys` from the store, or delete the entire store
+   * if no keys are passed. A `del` event is also emitted for each key
+   * deleted.
+   *
+   * **Note that to delete the entire store you must pass `{force: true}`**
+   *
+   * ```js
+   * store.del();
+   *
+   * // to delete paths outside cwd
+   * store.del({force: true});
+   * ```
+   *
+   * @param {String|Array|Object} `keys` Keys to remove, or options.
+   * @param {Object} `options`
+   * @api public
+   */
+
+  del(key) {
+    if (!key) key = this.keys;
+    if (Array.isArray(key)) {
+      for (const k of key) this.del(k);
+      return this;
+    }
+    assert.equal(typeof key, 'string', 'expected key to be a string');
+    delete this.data[key];
```

Comment:
This will only delete the top level keys. I think this is how it was before, but should we consider deleting nested keys?